lib/improver/wxcode/weather_symbols.py

```diff
@@ -257,21 +271,32 @@ def construct_extract_constraint(diagnostics, thresholds):
                 A thresholds within the given diagnostic cubes that are needed.
                 Including units.
         Returns:
-            iris.Constraint or list of iris.Constraints:
-                The constructed iris constraints.
+            string or list of strings:
+                String, or list of strings, encoding iris cube constraints.
         """
+        def _constraint_string(diagnostic, threshold):
+            """
+            Return iris constraint as a string for deferred creation of the
+            lambda functions.
+            Args:
+                diagnostic (string)
+                threshold (float)
+            Returns: (string)
+            """
+            return '{}{}{}{}{}{}{}'.format(
```

Comment:
I'm not sure why you've done it like this rather than something like:
`"iris.Constraint(name='{diagnostic}', threshold=lambda cell:  {threshold}*0.99 < cell < {threshold}*1.01)".format(diagnostic=diagnostic, threshold=threshold)`

This might make it clearer what you are putting into the constraint.

Reply:
Yeah, I must have been having a funny turn.